pkg/apis/ceph.rook.io/v1/types.go

```diff
@@ -2567,6 +2567,26 @@ type StorageScopeSpec struct {
 	// +nullable
 	// +optional
 	StorageClassDeviceSets []StorageClassDeviceSet `json:"storageClassDeviceSets,omitempty"`
+	// +optional
+	Store OSDStore `json:"store,omitempty"`
+}
+
+// StoreType is the backend storage type for OSDs
+type StoreType string
+
+const (
+	// Bluestore osd storage
+	BlueStore StoreType = "bluestore"
+	// Bluestore RDR storage
+	BlueStoreRDR StoreType = "bluestore-rdr"
+)
+
+// OSDStore is the backend storage type used for creating the OSDs
+type OSDStore struct {
+	// Type of backend storage to be used while creating OSDs. If empty, then bluestore will be used
+	// +optional
+	// +kubebuilder:validation:Enum=bluestore;bluestore-rdr;
```

Comment:
Have you verified if k8s allows you to have this be empty without `""` in the enum?

Reply:
Yes. It does allow me to send empty `store` without having `""` in the Enum
![Screenshot from 2023-07-17 11-43-30](https://github.com/rook/rook/assets/9363998/2c1411a0-f4b8-4852-9972-939c4469e7e6)